src/rdkafka.c

```diff
@@ -3238,16 +3292,62 @@ rd_kafka_resp_err_t rd_kafka_consumer_close(rd_kafka_t *rk) {
                 }
         }
 
+        if (rebalance_rkq) {
+                /* Wake up the application-provided rebalance queue so the
+                 * application can exit its poll loop. */
+                rd_kafka_q_yield(rebalance_rkq);
+
+                /* Drop our reference */
+                rd_kafka_q_destroy(rebalance_rkq);
+        }
+
+        /* Reset cgrp queue forwarding as no further events will/must be
+         * emitted. */
         rd_kafka_q_fwd_set(rkcg->rkcg_q, NULL);
```

Comment:
i don't understand how the above code blocks until here until the consumer has closed in the rebalance_rkq case. i suspect you know what you are doing and i'm missing something obvious, but you might want to a comment about this.